frontend/src/types/roles/role-assignment-status.ts

```diff
@@ -0,0 +1,7 @@
+export enum DecisionStatus {
+    Approved = 'approved',
+    Pending = 'pending',
+    Denied = 'denied',
+}
+
+export type DecisionStatusType = DecisionStatus.Approved | DecisionStatus.Pending | DecisionStatus.Denied;
```

Comment:
Typescipt enums are not so nice, you can take the suggested approach instead.
Reference: https://maxheiber.medium.com/alternatives-to-typescript-enums-50e4c16600b1

```suggestion
export const DecisionStatus = {
    APPROVED: 'approved',
    PENDING: 'pending',
    DENIED: 'denied',
} as const;

export type DecisionStatus = (typeof DecisionStatus)[keyof typeof DecisionStatus];
```

Reply:
Good to know, thank you for the info